packages/rum-recorder/src/domain/rrweb/record.spec.ts

```diff
@@ -1,40 +1,296 @@
 import { createNewEvent, isIE } from '@datadog/browser-core'
-import { RecordType, Record } from '../../types'
+import { serializedNodeWithId, NodeType } from 'rrweb-snapshot'
+import { collectAsyncCalls } from '../../../test/utils'
+import { Record, RecordType, IncrementalSource, MutationData, FullSnapshotRecord } from '../../types'
 import { record } from './record'
 
+// Each full snapshot is generating two records, a Meta record and a FullSnapshot record
+const RECORDS_PER_FULL_SNAPSHOTS = 2
+
 describe('record', () => {
+  let sandbox: HTMLElement
   let input: HTMLInputElement
   let stop: (() => void) | undefined
+  let emitSpy: jasmine.Spy<(record: Record) => void>
+  let waitEmitCalls: (expectedCallsCount: number, callback: () => void) => void
+  let expectNoExtraEmitCalls: (done: () => void) => void
 
   beforeEach(() => {
     if (isIE()) {
       pending('IE not supported')
     }
 
+    emitSpy = jasmine.createSpy()
+    ;({ waitAsyncCalls: waitEmitCalls, expectNoExtraAsyncCall: expectNoExtraEmitCalls } = collectAsyncCalls(emitSpy))
+    sandbox = document.createElement('div')
+    sandbox.id = 'sandbox'
     input = document.createElement('input')
-    document.body.appendChild(input)
+    sandbox.appendChild(input)
+    document.body.appendChild(sandbox)
   })
 
   afterEach(() => {
-    input.remove()
+    jasmine.clock().uninstall()
+    sandbox.remove()
     if (stop) {
       stop()
     }
   })
 
   it('will only have one full snapshot without checkout config', () => {
-    const emit = jasmine.createSpy<(record: Record) => void>()
-    stop = record<Record>({ emit })?.stop
+    stop = record<Record>({ emit: emitSpy })?.stop
+
+    const typeEventCount = 30
+    type(typeEventCount)
```

Comment:
nit: something like `dispatchInputEvents` instead of `type` could clarify the intent